roles/pre-ansimail/files/ansimail_scripts/virtual-regen

```diff
@@ -4,37 +4,37 @@
 
 . /usr/local/lib/ansimail/helpers/variables
 
-virtuals=$user_home/virtual
-revirt=$user_home/revirt
-bak=$( date +%Y%d%m%H%M )
+virtuals="$user_home/virtual"
+revirt="$user_home/revirt"
+bak=$(date +%Y%d%m%H%M)
 
-if [ -e $virtuals ]; then
-	mv $virtuals "$virtuals.$bak"
+if [ -e "$virtuals" ]; then
+	mv "$virtuals" "$virtuals.$bak"
 fi
-if [ -e $reivrt ]; then
-	mv $revirt "$revirt.$bak"
+if [ -e "$reivrt" ]; then
+	mv "$revirt" "$revirt.$bak"
 fi
 
-touch $virtuals $revirt
+touch "$virtuals" "$revirt"
 
 # Now check that every alias is only an alias
 # (this also ensures that every use is only a user, due to bipartite-ness)
-for user_file in $user_home/user-data/*; do
-	username=$( basename $user_file )
-	echo "\n$username\t\t\t\tansimail" >> $virtuals
-	printf "$username\t\t\t\t$username" >> $revirt
+for user_file in "$user_home/user-data/"*; do
+	username=$( basename "$user_file" )
+	printf "\n%s\t\t\t\tansimail" "$username" >> "$virtuals"
+	printf "%s\t\t\t\t%s" "$username" "$username" >> "$revirt"
 	while IFS= read -r alias; do
-		echo "$alias\t\t\t\t$username" >> $virtuals
-		printf ",$alias" >> $revirt
-	done < $user_home/user-data/$username
-	printf "\n" >> $revirt
+		printf "%s\t\t\t\t%s" "$alias" "$username" >> "$virtuals"
+		printf ",%s", "$alias" >> $revirt
```

Comment:
There seems to be an extra comma here in the `printf` after ",%s"

Should it be existing? I feel that it is a small typo, given you haven't added it anywhere else.

Reply:
The comma was there when I started I believe so I left it thinking you had it in there for some reason. I'll remove it.